<launch>

  <arg name="local_costmap_dir" default="params/map_nav_params/local_costmap_params.yaml"/>
  <arg name="global_costmap_dir" default="params/map_nav_params/global_costmap_params.yaml"/>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find nu_jackal_autonav)/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find nu_jackal_autonav)/params/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find nu_jackal_autonav)/$(arg local_costmap_dir)" command="load" />
    <rosparam file="$(find nu_jackal_autonav)/$(arg global_costmap_dir)" command="load" />
    <rosparam file="$(find nu_jackal_autonav)/params/base_local_planner_params.yaml" command="load" />
    <rosparam file="$(find nu_jackal_autonav)/params/move_base_params.yaml" command="load" />

    <!-- ALL OF THE GLOBAL PLUGINS AVAILABLE (choose 1) -->
    <!-- NOTE: JACKAL ONLY HAS navfn INSTALLED RIGHT NOW-->
    <param name="base_global_planner" type="string" value="navfn/NavfnROS" />
    <!-- <param name="base_global_planner" value="global_planner/GlobalPlanner" /> -->
    <!-- <param name="base_global_planner" value="carrot_planner/CarrotPlanner"/> -->

    <!-- ALL OF THE LOCAL PLUGINS AVAILABLE (choose 1) -->
    <!-- NOTE: JACKAL ONLY HAS TrajectoryPlannerROS INSTALLED RIGHT NOW -->
    <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS"/>
    <!-- <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS"/> -->
    <!-- <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/> -->
    <!-- <param name="base_local_planner" value="custom_planner/CustomPlanner"/> -->
    <remap from="odom" to="odometry/corrected"/>
    <remap from="cmd_vel" to="cmd_vel/nav"/>
  </node>

</launch>
